crates/spicepod/src/component/dataset.rs

```diff
@@ -230,6 +247,9 @@ pub mod acceleration {
 
         #[serde(default, skip_serializing_if = "Option::is_none")]
         pub primary_key: Option<String>,
+
+        #[serde(default, skip_serializing_if = "HashMap::is_empty")]
+        pub on_conflict: HashMap<String, OnConflictBehavior>,
```

Comment:
This is where the Spicepod definition is defined. This unlocks the following syntax:

```yaml
on_conflict:
  hash: drop
```

```yaml
on_conflict:
  "(hash, log_index)": upsert
```

Note that the Spicepod definition just treats the key as a String - we will parse this into a column reference later on.